apps/steercode-app-electron/src/renderer/src/lib/shared/stores/user.ts

```diff
@@ -0,0 +1,30 @@
+import type { User } from '$lib/models/types/user.type';
+import { writable } from 'svelte/store';
+
+import {
+  customFetch,
+  responseWithErrorHandeling
+} from '$lib/core/services/request';
+
+function createUserStore() {
+  const user = writable<User | null>(null);
+
+  async function fetchUserInfo(uid: string) {
+    const response = customFetch(`/user/`);
+    const userInfo = await responseWithErrorHandeling<User | null>(
+      response,
+      null
+    );
+    if (userInfo) {
+      user.set(userInfo);
+    }
+  }
+
+  return {
+    subscribe: user.subscribe,
+    set: user.set,
```

Comment:
set function should not be exported from the userStore. Setting the user outside of this store might cause unpredictable behavior.